app/assets/javascripts/admin/addon/components/color-palette-editor.gjs

```diff
@@ -0,0 +1,186 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import concatClass from "discourse/helpers/concat-class";
+import dIcon from "discourse/helpers/d-icon";
+import { i18n } from "discourse-i18n";
+
+const LIGHT = "light";
+const DARK = "dark";
+
+class Color {
+  @tracked lightValue;
+  @tracked darkValue;
+
+  constructor({ name, lightValue, darkValue }) {
+    this.name = name;
+    this.lightValue = lightValue;
+    this.darkValue = darkValue;
+  }
+
+  get displayName() {
+    return this.name.replaceAll("_", " ");
+  }
+
+  get description() {
+    return i18n(`admin.customize.colors.${this.name}.description`);
+  }
+}
+
+const NavTab = <template>
+  <li>
+    <a
+      class={{concatClass "" (if @active "active")}}
+      tabindex="0"
+      {{on "click" @action}}
+      {{on "keydown" @action}}
+      ...attributes
+    >
+      {{dIcon @icon}}
+      <span>{{@label}}</span>
+    </a>
+  </li>
+</template>;
+
+const Picker = class extends Component {
+  @action
+  onInput(event) {
+    const color = event.target.value.replace("#", "");
+    if (this.args.showDark) {
+      this.args.color.darkValue = color;
+    } else {
+      this.args.color.lightValue = color;
+    }
+  }
+
+  @action
+  onChange(event) {
+    const color = event.target.value.replace("#", "");
+    if (this.args.showDark) {
+      this.args.onDarkChange(color);
```

Comment:
Do I undestand it correctly it is used to trigger events when color changes. Currently it is only used in test but it will probably be useful for preview?

Reply:
Yup, it's not used anywhere in the app yet, but it will be as we start building out the new admin color palette UI.